cmd/juju/commands/debugcode.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2012, 2013 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package commands
+
+import (
+	"github.com/juju/cmd"
+	"github.com/juju/errors"
+	"github.com/juju/gnuflag"
+	"gopkg.in/juju/names.v3"
+
+	jujucmd "github.com/juju/juju/cmd"
+	"github.com/juju/juju/cmd/modelcmd"
+	"github.com/juju/juju/network/ssh"
+)
+
+func newDebugCodeCommand(hostChecker ssh.ReachableChecker) cmd.Command {
+	c := new(debugCodeCommand)
+	c.getActionAPI = c.debugHooksCommand.newActionsAPI
+	c.setHostChecker(hostChecker)
+	return modelcmd.Wrap(c)
+}
+
+// debugCodeCommand connects via SSH to a running unit, and drops into a tmux shell,
+// prepared to debug hooks/actions as they fire.
+type debugCodeCommand struct {
+	debugHooksCommand
+	debugAt string
+}
+
+const debugCodeDoc = `
+Interactively debug hooks and actions on a unit.
+
+Similar to 'juju debug-hooks' but rather than dropping you into a shell prompt, 
+it runs the hooks and sets the JUJU_DEBUG_AT environment variable. 
+Charms that implement support for this should use it to set breakpoints based on the environment
+variable.
+
+See the "juju help ssh" for information about SSH related options
+accepted by the debug-hooks command.
+`
+
+func (c *debugCodeCommand) Info() *cmd.Info {
+	return jujucmd.Info(&cmd.Info{
+		Name:    "debug-code",
```

Comment:
Given how long it took us to get to 'debug-code', I'm not going to change it in this PR, but since that wasn't a name we discussed, I'll bring it up for discussion and we can change it in a future PR.